syntax = "proto3";
option java_package = "com.seatunnel.grpc";
option java_multiple_files = true;
package seatunnel.rpc;

service MetaStreamService {
  rpc DataSetDataStream(stream MetaDataSetDataStream) returns (Empty);
  rpc FetchData(MetaDataSetDataStream) returns (stream MetaDataSetDataStream);
}

message Empty{

}

message MetaDataSetDataStream{
    string data_set_id = 1;
    string trace_id = 2;
    map<string, Project> data = 3;
}

enum VarType {
  INT32 = 0;
  INT64 = 1;
  STRING = 2;
  FLOAT = 3;
  DOUBLE = 4;
  BOOL = 5;
  BYTE = 6;
  TIMESTAMP = 7;
}

message Project {
    VarType var_type = 1;
    oneof oneof_value {
        int32 value_int32 = 2;
        bytes value_string = 3;
        int64 value_int64 = 4;
        float value_float = 5;
        double value_double = 6;
        bool value_bool = 7;
        bytes value_bytes = 8;
    }
}


